packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -222,6 +229,7 @@ export function useAsyncData<
     if (fetchOnServer && nuxt.isHydrating && hasCachedData()) {
       // 1. Hydration (server: true): no fetch
       asyncData.pending.value = false
+      asyncData.status.value = asyncData.error.value ? 'error' : 'success'
```

Comment:
would it make sense instead to initialise it to this value above on L132?

Reply:
Just to explain, I wasn't thinking we should reset to idle but rather could directly initialise it to 'error' or 'success' based on the value in the payload _rather_ than initialising it to 'idle'.

But on reflection, I think this is good too 👍

Reply:
I misunderstood, gotcha! 👍